src/components/Settings/SettingsDialog.tsx

```diff
@@ -41,17 +42,12 @@ const tabs = [
 ]
 
 export const SettingsDialog = ({ open, onOpenChange }: SettingsDialogProps) => {
-  const [settings, setSettings] = useState<AppSettings>()
-  const [submitting, setSubmitting] = useState(false)
+  const { data: settings } = useSettings()
+  const { mutateAsync: saveSettings, isPending } = useSaveSettings(() => {
+    onOpenChange(false)
+  })
```

Comment:
It's a bit weird, because of how the save handler in main doesn't throw an error, but returns `false` instead. This means that a save error will still be considered "success", but at the same time we don't want the dialog to close if it wasn't actually saved. 

I think the handler should just re-throw the error, but I also don't want to make any changes to it now, because we have the same pattern in a few other places :\

Reply:
I see. Well, you have my blessing! 👍